zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/ProcessInstanceControllerTest.java

```diff
@@ -0,0 +1,587 @@
+        {
+            "bpmnProcessId": "bpmnProcessId",
+            "version": 1,
+            "awaitCompletion": true
+        }""";
+
+    // when / then
+    webClient
+        .post()
+        .uri(PROCESS_INSTANCES_START_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_START_RESPONSE);
+
+    verify(processInstanceServices).startProcessInstanceWithResult(requestCaptor.capture());
+    final var capturedRequest = requestCaptor.getValue();
+    assertThat(capturedRequest.bpmnProcessId()).isEqualTo("bpmnProcessId");
+    assertThat(capturedRequest.version()).isEqualTo(1);
+  }
+
+  @Test
+  void shouldRejectStartProcessInstancesIfNoBpmnProcessIdOrProcessDefinitionKeyAreProvided() {
+    // given
+    final var request =
+        """
+        {
+            "version": 1,
+            "awaitCompletion": true
+        }""";
+
+    final var expectedBody =
+        """
+        {
+            "type":"about:blank",
+            "title":"INVALID_ARGUMENT",
+            "status":400,
+            "detail":"At least one of [bpmnProcessId, processDefinitionKey] is required.",
+            "instance":"/v2/process-instances"
+         }""";
+
+    // when / then
+    webClient
+        .post()
+        .uri(PROCESS_INSTANCES_START_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_PROBLEM_JSON)
+        .expectBody()
+        .json(expectedBody);
+  }
+
+  @Test
+  void shouldRejectStartProcessInstancesIfBothBpmnProcessIdOrProcessDefinitionKeyAreProvided() {
+    // given
+    final var request =
+        """
+        {
+            "bpmnProcessId": "bpmnProcessId",
+            "processDefinitionKey": 123,
+            "version": 1
+        }""";
+
+    final var expectedBody =
+        """
+        {
+            "type":"about:blank",
+            "title":"INVALID_ARGUMENT",
+            "status":400,
+            "detail":"Only one of [bpmnProcessId, processDefinitionKey] is required.",
+            "instance":"/v2/process-instances"
+         }""";
+
+    // when / then
+    webClient
+        .post()
+        .uri(PROCESS_INSTANCES_START_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_PROBLEM_JSON)
+        .expectBody()
+        .json(expectedBody);
+  }
+
+  @Test
+  void shouldRejectStartProcessInstancesIfBpmnProcessIdIsProvidedButNoVersion() {
```

Comment:
🔧 Looking at the [processor in the broker](https://github.com/camunda/camunda/blob/main/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceCreationCreateProcessor.java#L337-L342), it is OK to provide a `bpmnProcessId` and no `version`.

Reply:
I misunderstood the internal doc, thanks for raising 👍